### PR TITLE
Custom ppl opts

### PR DESCRIPTION
**Pull request by yerbol-akhmetov**
## Changes proposed in this Pull Request
Here I propose an extension of functionality of custom powerplants addition. Currently, there are only two options: 1) Use open-source powerplants with `custom_powerplants: false`, and 2) Attach custom powerplants from `custom_powerplants.csv` to open-source data and use both with `custom_powerplants: true`. There is no option to select only *custom powerplants*. This functionality is very useful. Sometimes you can have an accurate custom powerplants information and you would like to replace retrieved open-source plants with custom data. On the other situations, you might need to use both open-source and custom powerplants. 
Therefore, I propose to extend the functionality to three options: 
1) `false` - no custom power plants are used;
2) `merge` - both open-source and custom powerplants from `custom_powerplants.csv` are merged and used;
3) `replace` - only custom powerplants from `custom_powerplants.csv` are used.

## Checklist

- [x] I consent to the release of this PR's code under the AGPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
- [x] I tested my contribution locally and it seems to work fine.
- [x] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

File: config.default.yaml [hunk 1]
```diff
@@ -142,7 +142,7 @@ electricity:
     Link: []  # H2 pipeline
 
   powerplants_filter: (DateOut >= 2022 or DateOut != DateOut)
-  custom_powerplants: false
+  custom_powerplants: false  # "false" - no custom powerplants, "merge" uses both open-source and custom powerplants, "replace" uses only custom powerplants
```
Comment (pz-max) [inline config.default.yaml]: Could you change the comment to wherever possible:
`#  "false" use only powerplantmatching (ppm) data, "merge" combines ppm and custom powerplants, "replace" use only custom powerplants`

File: scripts/build_powerplants.py [hunk 2]
```diff
@@ -247,9 +247,12 @@ def add_custom_powerplants(ppl, inputs, config):
     # if isinstance(custom_ppl_query, str):
     #     add_ppls.query(custom_ppl_query, inplace=True)
```
Comment (pz-max) [inline scripts/build_powerplants.py]: I know pypsa-eur is using line 247 and 248. Can you figure out its purpose?
I think your merge and replace option could make lines 247 and 248 not necessary. Maybe we can remove the comment?

**Review by pz-max — CHANGES_REQUESTED** [review]
Nice PR! :100: Some minor changes are required.
Could you also adjust the configuration.rst accordingly?

File: scripts/build_powerplants.py (hunk 2 above)
Comment (yerbol-akhmetov) [reply scripts/build_powerplants.py]: I have checked the code and it seems that it is left to enable some query to custom powerplants being added. We can leave them as commented or we can remove them.

Comment (yerbol-akhmetov) [reply scripts/build_powerplants.py]: But it is better to remove them, since if enabled it might contradict with the proposed PR.

**Review by pz-max — APPROVED** [review]

**Review by pz-max — APPROVED** [review]
Looks great. I just had to change the release note position

**Pull request merged**